Publishing the source code under MIT licence

poosh opened the pull request:
Due to licensing concerns and cross-license compatibility between various projects, I decided to re-publish the code under the MIT license. To do that, we need the approval of _every_ contributor to KF-ScrnBalance.

**By approving this pull request, you give your permission to release your contribution under MIT.**

Shtoyan commented:
Hi. This is your project, and you can do whatever you want with it. If you think it'll be better with a different license, then go for it.  

But if you ask for my opinion - it won't change anything. There have been hundreds of scrn servers with modifications, and they didn't contribute back to you / release the modifications / used your mod because of licensing concerns. In the last 10 years, I can count fewer than 10-15 people who actually contributed to "KF open source", gave credit to the original authors, and cared about licenses. The other hundreds - or even thousands - just don't care about that kind of thing.  

And they are the reason I/InsultingPros don't release `unrealscript` mods under permissive licenses. I gladly copy-paste `gpl3` text into all sources to emphasise that they're not doing the right thing.

poosh commented:
> Hi. This is your project, and you can do whatever you want with it. If you think it'll be better with a different license, then go for it.
> 
> But if you ask for my opinion - it won't change anything. There have been hundreds of scrn servers with modifications, and they didn't contribute back to you / release the modifications / used your mod because of licensing concerns. In the last 10 years, I can count fewer than 10-15 people who actually contributed to "KF open source", gave credit to the original authors, and cared about licenses. The other hundreds - or even thousands - just don't care about that kind of thing.
> 
> And they are the reason I/InsultingPros don't release `unrealscript` mods under permissive licenses. I gladly copy-paste `gpl3` text into all sources to emphasise that they're not doing the right thing.


I completely agree with you. The selfish, ignorant, and narrow-minded attitude of most KF modders, such as "I fixed five lines of code in your project; now everybody should ignore the original project and join my super server instead," hurts the KF community in general.

Let's take Stalker (sorry, S.T.A.L.K.E.R.) as an example, where people usually take other people's work, enhance it, and contribute back. For instance, Anomaly incorporated the Gunslinger mod, modified it, and submitted the changes back to the original repo. OGSR, which also uses Gunslinger, benefited from Anomaly, albeit both (OGSR and Anomaly) have nothing in common.

I'm stepping away from GPL for a different reason. Those who violate license agreements do not care anyway, while GPL may alienate those who do. For instance, KFTurbo was released under GPL 2.0, while ScrnBalance uses GPL 3.0. Since those GPL versions are incompatible, we cannot legally share the code between two projects. I signed up for GPL with "freedom" in mind. I didn't want a legal burden. Hence, I'm here to step away.

Moreover, I'm unsure if GPL is compatible with KF EULA.
